packages/amplify_datastore/example/integration_test/model_type_test.dart

```diff
@@ -0,0 +1,426 @@
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:amplify_datastore/amplify_datastore.dart';
+
+import 'package:amplify_datastore_example/models/ModelProvider.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'utils/model_test_operation_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('a model with field of type', () {
+    group('String', () {
+      var models = List.generate(5, (i) => StringTypeModel(value: '$i'));
+      testModelOperations<StringTypeModel>(
+        models: models,
+        classType: StringTypeModel.classType,
+      );
+    });
+
+    group('String (with null value)', () {
+      var models = List.generate(5, (i) => StringTypeModel());
+      testModelOperations<StringTypeModel>(
+        models: models,
+        classType: StringTypeModel.classType,
+      );
+    });
+
+    group('List<String>', () {
+      var list = List.generate(3, (i) => '$i');
+      var models = List.generate(5, (_) => StringListTypeModel(value: list));
+      testModelOperations<StringListTypeModel>(
+        models: models,
+        classType: StringListTypeModel.classType,
+      );
+    });
+
+    group('List<String> (with null value)', () {
+      var models = List.generate(5, (_) => StringListTypeModel());
+      testModelOperations<StringListTypeModel>(
+        models: models,
+        classType: StringListTypeModel.classType,
+      );
+    });
+
+    group('int', () {
+      var models = List.generate(5, (i) => IntTypeModel(value: i));
+      testModelOperations<IntTypeModel>(
+        models: models,
+        classType: IntTypeModel.classType,
+      );
+    });
+
+    group('int (with null value)', () {
+      var models = List.generate(5, (i) => IntTypeModel());
+      testModelOperations<IntTypeModel>(
+        models: models,
+        classType: IntTypeModel.classType,
+      );
+    });
+
+    group('List<int>', () {
+      var list = List.generate(3, (i) => i);
+      var models = List.generate(5, (_) => IntListTypeModel(value: list));
+      testModelOperations<IntListTypeModel>(
+        models: models,
+        classType: IntListTypeModel.classType,
+      );
+    });
+
+    group(
+      'List<int> (with null value)',
+      () {
+        var models = List.generate(5, (_) => IntListTypeModel());
+        testModelOperations<IntListTypeModel>(
+          models: models,
+          classType: IntListTypeModel.classType,
+        );
+      },
+      // unskip when https://github.com/aws-amplify/amplify-flutter/issues/813 is resolved
```

Comment:
TODO is used differently by different devs, but IMO TODO should be used to mark something you intend to fix prior to merge. When they get merged into the code base they tend to stay there for an eternity and make it harder to find your own TODOs that you actually intend to fix before merging your work. For example, the branch I am on now has 20 TODOs, some of which were added by me and I intend to fix prior to creating a PR, but its hard to tell which is which.

Reply:
I meant to add this - Chris asked how we would make sure these actually get updated. I will add a link to each issue once these are merged so that we see them in the issue.

Reply:
>TODO should be used to mark something you intend to fix prior to merge

I don't think that's the case?

`TODO` should help to locate (via an IDE) all items in the codebase that need to follow up in general, these items might not able to be resolved before merge due to various reasons. Could that be easier than searching keyword or looking for issue links?